go-controller/pkg/ovn/pods_test.go

```diff
@@ -231,12 +239,7 @@ var _ = Describe("OVN Pod Operations", func() {
 				Expect(err).NotTo(HaveOccurred())
 				Eventually(fExec.CalledMatchesExpected).Should(BeTrue(), fExec.ErrorDesc)
 
-				pod, err = fakeOvn.fakeClient.CoreV1().Pods(t.namespace).Get(context.TODO(), t.podName, metav1.GetOptions{})
-				Expect(err).NotTo(HaveOccurred())
-
-				podAnnotation, ok := pod.Annotations[util.OvnPodAnnotationName]
-				Expect(ok).To(BeTrue())
-				Expect(podAnnotation).To(MatchJSON(`{"default": {"ip_addresses":["` + t.podIP + `/24"], "mac_address":"` + t.podMAC + `", "gateway_ips": ["` + t.nodeGWIP + `"], "ip_address":"` + t.podIP + `/24", "gateway_ip": "` + t.nodeGWIP + `"}}`))
+				Eventually(getPodAnnotations(fakeOvn.fakeClient, t.namespace, t.podName), 2).Should(MatchJSON(`{"default": {"ip_addresses":["` + t.podIP + `/24"], "mac_address":"` + t.podMAC + `", "gateway_ips": ["` + t.nodeGWIP + `"], "ip_address":"` + t.podIP + `/24", "gateway_ip": "` + t.nodeGWIP + `"}}`))
```

Comment:
This has the same problem as in the other bug; you are passing a constant value to `Eventually`, not passing a function for it to run repeatedly, so it is still just requiring the annotation to be set immediately.

Reply:
Gah you're right, it needs to be:

```
Eventually(func() string { return getPodAnnotations(fakeOvn.fakeClient, t.namespace, t.podName) }, 2).Should(MatchJSON(...))
```

Reply:
:facepalm: